src/cpp/CMakeLists.txt

```diff
@@ -355,6 +361,110 @@ else()
+       list(APPEND SOCI_LIBRARIES "${RSTUDIO_WINDOWS_DEPENDENCIES_DIR}/install-soci/postgresql/lib/libpq-debug-${SOCI_ARCH}.lib")
+       list(APPEND SOCI_LIBRARIES "${RSTUDIO_WINDOWS_DEPENDENCIES_DIR}/install-soci/postgresql/lib/libpgcommon-debug-${SOCI_ARCH}.lib")
+       list(APPEND SOCI_LIBRARIES "${RSTUDIO_WINDOWS_DEPENDENCIES_DIR}/install-soci/postgresql/lib/libpgport-debug-${SOCI_ARCH}.lib")
+   else()
+       file(GLOB_RECURSE SOCI_LIBRARIES "${SOCI_LIBRARY_DIR}/Release/*.lib")
+       list(APPEND SOCI_LIBRARIES "${WIN_LDAP_LIBRARY}")
+       list(APPEND SOCI_LIBRARIES "${WIN_SSPI_LIBRARY}")
+       list(APPEND SOCI_LIBRARIES "${RSTUDIO_WINDOWS_DEPENDENCIES_DIR}/install-soci/sqlite/sqlite3-release-${SOCI_ARCH}.lib")
+       list(APPEND SOCI_LIBRARIES "${RSTUDIO_WINDOWS_DEPENDENCIES_DIR}/install-soci/postgresql/lib/libpq-release-${SOCI_ARCH}.lib")
+       list(APPEND SOCI_LIBRARIES "${RSTUDIO_WINDOWS_DEPENDENCIES_DIR}/install-soci/postgresql/lib/libpgcommon-release-${SOCI_ARCH}.lib")
+       list(APPEND SOCI_LIBRARIES "${RSTUDIO_WINDOWS_DEPENDENCIES_DIR}/install-soci/postgresql/lib/libpgport-release-${SOCI_ARCH}.lib")
+   endif()
+endif()
+list(LENGTH SOCI_LIBRARIES SOCI_LIB_COUNT)
+
+if (SOCI_LIB_COUNT EQUAL 0)
+   message(FATAL_ERROR "No SOCI libraries found under ${SOCI_LIBRARY_DIR}. Ensure the SOCI dependency is installed and try again.")
+else()
+   message(STATUS "SOCI libraries found under ${SOCI_LIBRARY_DIR}.")
+
+   # ensure the soci/sqlite/postgres libraries are installed with the installation package
+   if(UNIX)
+      install(PROGRAMS "${SOCI_LIBRARY_DIR}/libsoci_core${SOCI_LIB_SUFFIX}" DESTINATION ${RSTUDIO_INSTALL_BIN})
+      install(PROGRAMS "${SOCI_LIBRARY_DIR}/libsoci_sqlite3${SOCI_LIB_SUFFIX}" DESTINATION ${RSTUDIO_INSTALL_BIN})
+      install(PROGRAMS "${SOCI_LIBRARY_DIR}/libsoci_postgresql${SOCI_LIB_SUFFIX}" DESTINATION ${RSTUDIO_INSTALL_BIN})
+
+      GET_PREREQUISITES("${SOCI_LIBRARY_DIR}/libsoci_sqlite3${SOCI_LIB_SUFFIX}" SQLITE_LIBS "0" "0" "." "")
+      foreach(LIBRARY IN LISTS SQLITE_LIBS)
+         string(FIND "${LIBRARY}" "libsqlite3" INDEX)
+         if(INDEX GREATER "-1")
+            set(SQLITE_LIB "${LIBRARY}")
+         endif()
+      endforeach(LIBRARY)
+
+      GET_PREREQUISITES("${SOCI_LIBRARY_DIR}/libsoci_postgresql${SOCI_LIB_SUFFIX}" POSTGRES_LIBS "0" "0" "." "")
+      foreach(LIBRARY IN LISTS POSTGRES_LIBS)
+         string(FIND "${LIBRARY}" "libpq" INDEX)
+         if(INDEX GREATER "-1")
+            set(POSTGRES_LIB "${LIBRARY}")
+         endif()
+      endforeach(LIBRARY)
+
+      get_filename_component(SQLITE_LIB "${SQLITE_LIB}" REALPATH)
+      get_filename_component(POSTGRES_LIB "${POSTGRES_LIB}" REALPATH)
+      message(STATUS "SQLite lib: ${SQLITE_LIB}")
+      message(STATUS "PostgreSQL lib: ${POSTGRES_LIB}")
+      install(PROGRAMS "${SQLITE_LIB}" DESTINATION ${RSTUDIO_INSTALL_BIN})
```

Comment:
Do we need to add something to NOTICE for libsqlite?

Reply:
No, it's actually public domain!